microsoft/terminal
# Immediately flush a frame when we encounter a string we don't understand

## Summary of the Pull Request

Whenever conpty encounters a VT sequence it doesn't understand, we'll make sure to flush the current frame before writing that string through to the connected terminal application. 

 
## References

This might have some long-term consequences for #1173. IIRC I wrote some code that was fairly similar for passthrough mode. This might be a subset of that code.


## PR Checklist
* [x] Closes #2011
* [n/a] Requires documentation to be updated


## Detailed Description of the Pull Request / Additional comments

We need to do this because some VT sequences might be dependent upon state that's currently buffered in the frame. So we'll flush the frame so the terminal also has that state, then we'll write the sequence we didn't understand/wanted to passthrough. 


## Validation Steps Performed

While testing this might be a bit trickier, I wrote a test script that changes the cursor color, in the middle of a big block of text. Changing the cursor color is a VT sequence we passthrough always. Before the change, the cursor color sequence would always appear _before_ any of the related text. After the change, the text before the color sequence is always emitted before the sequence, and the text after the sequence is always emitted after.
